core/unit_test/view/TestViewConversion.cpp

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright Contributors to the Kokkos project
+
+#include <Kokkos_Core.hpp>
+#include <type_traits>
+
+// Checking requirement of explict type conversion to View
+
+namespace {
+
+using T     = int;
+using ptr_t = T*;
+
+using view0_t = Kokkos::View<T>;
+using view1_t = Kokkos::View<T*>;
+using view2_t = Kokkos::View<T[4]>;
+
+static_assert(!std::is_convertible_v<ptr_t, view0_t>);
+static_assert(!std::is_convertible_v<ptr_t, view1_t>);
+static_assert(!std::is_convertible_v<ptr_t, view2_t>);
```

Comment:
Not blocking but IMO these aliases are pointless.
```suggestion
static_assert(!std::is_convertible_v<T*, Kokkos::View<T>>);
static_assert(!std::is_convertible_v<T*, Kokkos::View<T*>>);
static_assert(!std::is_convertible_v<T*, Kokkos::View<T[4]>>);
```

Reply:
Also this would be a better thing to check
```C++
template <class From, class To>
inline constexpr bool conversion_is_explicit =
    std::is_constructible_v<To, From> && !std::is_convertible_v<From, To>;
```

Reply:
I just wanted to make sure you also saw https://github.com/kokkos/kokkos/pull/8528#discussion_r2411845971